src/backend/optimizer/util/pathnode.c

```diff
@@ -1565,6 +1565,7 @@ set_append_path_locus(PlannerInfo *root, Path *pathnode, RelOptInfo *rel,
 		else
 		{
 			subpath = cdbpath_create_motion_path(root, subpath, subpath->pathkeys, false, targetlocus);
+			Insist(subpath);
```

Comment:
Why do we insist here that `subpath` should not be NULL? Do we promise somewhere in the code  that `subpath->pathkeys` is NIL? Of cause we can get a segfault in the next line if a new value of `subpath` is NULL, but may be we should handle it in some other way?

Reply:
This is not about `subpath->pathkeys`. Like in other cases, this call pass false to `require_existing_order`, which means it doesn't expect to get NULL output value.
What other ways of handling do you suggest?

Reply:
At the moment we declare another `cdbpath_create_motion_path()` convention:
- `require_existing_order == true` -> we **may** return NULL (not this case)
- `require_existing_order == false && pathkeys == NULL` -> we **always** return NULL (should insist subpath exists)

We have `require_existing_order == false && pathkeys == unknown`, so I think we should not insist but rather fail with a some other error.

Reply:
> * `require_existing_order == false && pathkeys == NULL` -> we **always** return NULL (should insist subpath exists)

Sorry, but I can't understand. We are talking about this condition, right?
```
if (require_existing_order &&
    !pathkeys)
    return NULL;
```
We return NULL if `require_existing_order == true && pathkeys == NULL`, not `require_existing_order == false` as you writing.
It means, getting NULL calling `cdbpath_create_motion_path` with the specified arguments is unexpected behavior, which is handled by `Insist`.